proc/variables.go

```diff
@@ -28,20 +28,47 @@ const (
 
 // Represents a variable.
 type Variable struct {
-	Addr      uintptr
-	Name      string
-	Value     string
-	Type      string
-	dwarfType dwarf.Type
-	thread    *Thread
-
-	Len       int64
-	Cap       int64
+	Addr       uintptr
+	Name       string
+	DwarfType  dwarf.Type
+	RealType   dwarf.Type
+	SimpleType SimpleType
+	thread     *Thread
+
+	Value interface{}
+
+	Len int64
+	Cap int64
+
 	base      uintptr
 	stride    int64
 	fieldType dwarf.Type
+
+	Children []Variable
+
+	loaded     bool
+	Unreadable error
 }
 
+type SimpleType string
+
+const (
```

Comment:
On second thought, instead of defining our own simple types, the Go linker actually provides this information in the Dwarf info, we just need to parse it: https://github.com/golang/debug/blob/97a893dc61d8b085786f92514092a127fe25589c/dwarf/const.go#L89. Once we have that value we can use `reflect.Kind` to get a string representing the Go type.